getting-started/testing-setup.md

```diff
@@ -0,0 +1,220 @@
+{% endhint %}
+
+Next, we will need to navigate to `config/application.rb` and customize it to look like the following:
+
+{% code title="config/application.rb" %}
+```ruby
+module AmazingStore
+  class Application < Rails::Application
+
+    # … everything else
+
+    # Don't initialize this if factory_bot_rails is not part of the current bundle group.
+    if defined?(FactoryBotRails)
+      initializer after: "factory_bot.set_factory_paths" do
+        require 'spree/testing_support'
+
+        # The paths for solidus factories.
+        solidus_paths = Spree::TestingSupport::FactoryBot.definition_file_paths
+
+        # Optional: The paths for extensions may be in different places, please refer to each extension
+        # and add the factories you require without the ".rb" extension, avoid the directory
+        # as it's supported by FactoryBot but can generate ambigous paths when a file with the
+        # same name exists.
+        extension_paths = []
+        # Optional:
+        # [ alt_1, alt_2, alt_3].map { |path| path.chomp(".rb") }
+
+        # Where:
+        # alt_1: MySolidusExtension::Engine.root.join("lib/solidus_content/factories.rb),
+        # alt_2: MySolidusExtension::Engine.root.join("lib/solidus_content/factories/product.rb"),
+        # alt_3: MySolidusExtension::Engine.root.join("lib/solidus_content/factories/product_factory.rb"),
+        # etc.
+
+        # The application factories, according to the place in which they're stored.
+        app_paths = [
+          Rails.root.join('lib', 'factories'),
+          Rails.root.join('spec', 'factories'),
+        ]
+
+        FactoryBot.definition_file_paths = solidus_paths + extension_paths + app_paths
+      end
+    end
+  end
+end
+```
+{% endcode %}
+
+Finally, modify the configuration and create the support file to reference customized methods. This will give you the ability to create/build factories and stub authentication:
+
+{% tabs %}
+{% tab title="spec/rails\_helper.rb" %}
+```ruby
+...
+# Add additional requires below this line. Rails is not loaded until this point!
+require 'spree/testing_support/authorization_helpers'
+require 'spree/testing_support/controller_requests'
+...
+# The following line is provided for convenience purposes. It has the downside
+# of increasing the boot-up time by auto-requiring all files in the support
+# directory. Alternatively, in the individual `*_spec.rb` files, manually
+# require only the support files necessary.
+Dir[Rails.root.join('spec','support','**','*.rb')].sort.each {|f| require f }
+...
+RSpec.configure do |config|
+  ...
+  config.include Spree::TestingSupport::ControllerRequests, spree_controller: true
+end
+```
+{% endcode %}
+{% endcode %}
+{% endtab %}
+
+{% tab title="spec/support/factory\_bot.rb" %}
+```ruby
+RSpec.configure do |config|
+  #...
+  config.include FactoryBot::Syntax::Methods
+end
+```
+{% endcode %}
+{% endtab %}
+{% endtabs %}
+
+## Byebug
```

Comment:
`byebug` is not related to testing, it is used in development as well. I think it shouldn't be part of this section.